### PR TITLE
Improve ResourceInfo

### LINKED ISSUE #84: Improve ResourceInfo fields
For example, 
participant -> externalIdentifier
uploadUrl -> remoteLocation,
resourceFolderRelatovePath -> localLocation,
fileType -> ContentType

### PR DESCRIPTION
**Pull request by MJ1998**
Fixes #84

### CODE REVIEW

File: sensing/src/main/java/com/google/android/sensing/db/impl/entities/ResourceInfoEntity.kt [hunk 1]
```diff
@@ -49,20 +50,26 @@ internal data class ResourceInfoEntity(
...
+  /** Label to display in place of the data. */
+  val resourceTitle: String,
 
-  /** Absolute and immutable upload url. */
-  val uploadURL: String,
+  /** Mime type of the content, with charset etc. */
+  val contentType: String,
+
+  /** Date this resource was first created. */
+  val creation: Instant,
```
Comment (chandrashekar-s) [inline sensing/src/main/java/com/google/android/sensing/db/impl/entities/ResourceInfoEntity.kt:68]: Can you please check if the `Instant` type is correctly handled by DB correctly.

File: sensing/src/main/java/com/google/android/sensing/db/impl/entities/ResourceInfoEntity.kt [hunk 2]
```diff
@@ -49,20 +50,26 @@ internal data class ResourceInfoEntity(
...
 
   /** Upload status. */
   val status: RequestStatus,
+
+  /** The last time the [status] was updated. */
+  val lastUpdateTime: Instant
```
Comment (chandrashekar-s) [inline sensing/src/main/java/com/google/android/sensing/db/impl/entities/ResourceInfoEntity.kt:74]: On update of the record, this value should also be updated to current Instant value..

**Review by chandrashekar-s — COMMENTED** [review]
Thanks for the changes @MJ1998, there are few minor comments. Can you please have a look.

File: sensing/src/main/java/com/google/android/sensing/db/impl/entities/ResourceInfoEntity.kt (hunk 1 above)
Comment (MJ1998) [reply sensing/src/main/java/com/google/android/sensing/db/impl/entities/ResourceInfoEntity.kt:68]: Yes it is! [DbTypeConverters](https://github.com/google-research/CVD-paper-mobile-camera-example/blob/main/sensing/src/main/java/com/google/android/sensing/db/impl/DbTypeConverters.kt) handle Instant type.

File: sensing/src/main/java/com/google/android/sensing/db/impl/entities/ResourceInfoEntity.kt (hunk 2 above)
Comment (MJ1998) [reply sensing/src/main/java/com/google/android/sensing/db/impl/entities/ResourceInfoEntity.kt:74]: Done.

**Review by chandrashekar-s — APPROVED** [review]
LGTM

**Review by chandrashekar-s — APPROVED** [review]

**Pull request merged**